src/algokit_utils/application_client.py

```diff
@@ -1278,8 +1261,10 @@ def execute_atc_with_logic_error(
         return atc.execute(algod_client, wait_rounds=wait_rounds)
     except Exception as ex:
         logic_error = _try_convert_to_logic_error(ex, approval_program, approval_source_map)
-        if logic_error:
-            raise logic_error from ex
+        if logic_error and config.debug:
+            simulate_response = atc.simulate(algod_client)
```

Comment:
We don't appear to be using the result of simulate to return any additional information to the user.